ARCs/arc-0060.md

```diff
@@ -0,0 +1,449 @@
+    "ARC60Domain": {
+      "type": "string",
+      "description": "The ARC-60 domain separator"
+    },
+    "bytes": {
+      "type": "array",
+      "items": {
+        "type": "integer",
+        "minimum": 0,
+        "maximum": 255
+      },
+      "description": "Challenge to be signed for authentication",
+      "maxItems": 32
+    }
+  },
+  "required": ["ARC60Domain", "bytes"]
+}
+```
+
+- The `ARC60Domain` object indicates the domain separator to be used for signing. It **SHOULD** be set to `"arc60"` or empty `""`. Algorand transaction domain separators `TX` and `TG` **MUST** be forbidden.
+- The `bytes` object is the byte array of data being signed. In the example, it is a 32-byte challenge to be signed for authentication scope. The `bytes` field cannot be a valid Algorand transaction, therefore it **MUST NOT** be prepended with a known domain separator `TX` or `TG`.
+
+> Algorand domain separators can be found in the <a href="https://github.com/algorandfoundation/specs/blob/master/dev/crypto.md#domain-separation">Algorand specs</a> and in the <a href="https://github.com/algorand/go-algorand/blob/master/protocol/hash.go#L21">Go type HashID</a>.
+
+A valid `StdDataStr` object for `ScopeType=AUTH` should look like the following:
+
+```json
+{
+  "ARC60Domain" : "arc60",
+  "bytes" : "[99, 50, 97, 57, 51, 99, ..., 57, 49, 53]"
+}
+```
+
+#### Error interface `SignDataError`
+
+The `SignDataError` object extends the `SignTxnsError` defined in [ARC-1](./arc-0001.md).
+
+```ts
+export interface SignDataError extends Error {
+  code: number;
+  data?: any;
+  failingSignData: (StdSigData | null);
+}
+```
+
+`SignDataError` uses the same error codes of `SignTxnsError` as well as the following codes:
+
+| Status Code | Name | Description |
+| --- | --- | --- |
+| 4600 | Invalid scope | The scope of the signing action is not recognized by the wallet |
+| 4601 | Invalid schema | The schema does not comply with ARC-60 requirements |
+| 4602 | Invalid encoding | Data cannot be decoded with the given encoding |
+
+### Semantic Requirements
+
+The call `signData(signingData, metadata)` **MUST** either return the signed data `ret` or reject the call throwing an error `err`.
+
+> Following [ARC-1](./arc-0001.md) terminology, in this ARC the term **Rejecting** means throwing an error with `err.code=4300`.
+
+Upon calling `signData(signingData,, metadata)`:
+
+- the `signingData` and `metadata` **MUST NOT** be `null`, otherwise the wallet **MUST** reject the call.
+- `signingData.data` **MUST** be valid respect the JSON schema in `metadata.schema`. If the validation fails, the call **MUST** be rejected with a `4300` error.
+- if the encoding `metadata.encoding` is present, it **MUST** be used to decode `signingData.data.bytes`.
+- the wallet **MUST** prepare the signing message as `msg=(<signingData.data.bytes>)`.
+  - if the value of `ARC60Domain` is not empty than it **MUST** be used as `prefix`: `msg=(<prefix> || <signingData.data.bytes>)`.
```

Comment:
Can we generalize / simplify this statement and just write:

"the wallet **MUST** prepare the signing message as `msg=(<signingData.data.ARC60Domain> || <signingData.data.bytes>)`."

If `ARC60Domain` is empty, it will anyway be equivalent to `msg=(<signingData.data.bytes>)`

Reply:
https://github.com/algorandfoundation/ARCs/pull/284/commits/4bad240350f70b931991c4765ea7ea0886999b52